test/drivers/test_cqrrp.cc

```diff
@@ -102,7 +102,7 @@ class TestCQRRP : public ::testing::Test
 
         ASSERT_NEAR(norm_AQR / norm_A,         0.0, atol);
         ASSERT_NEAR(max_col_norm / col_norm_A, 0.0, atol);
-        ASSERT_NEAR(norm_0 / std::sqrt((T) n), 0.0, atol);
+        ASSERT_LE(norm_0, atol * std::sqrt((T) n));
```

Comment:
My dude. You use the same ASSERT_NEAR pattern in the two lines directly above this. Please fix. To find other instances of this by looking for "``0.0, atol);``" throughout the codebase.

Reply:
Fixed.